engine/src/main/java/org/hibernate/validator/cfg/propertyholder/Constrainable.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.cfg.propertyholder;
+
+import org.hibernate.validator.cfg.ConstraintDef;
+
+/**
+ * Facet of a property holder constraint mapping creational context which allows to place
+ * constraints on the underlying element.
+ *
+ * @author Gunnar Morling
+ * @author Kevin Pollet &lt;kevin.pollet@serli.com&gt; (C) 2011 SERLI
+ */
+public interface Constrainable<C extends Constrainable<C>> {
```

Comment:
Same here about it being generic. Maybe `ConstrainableProperty` would be better?